src/peer.cpp

```diff
@@ -50,18 +53,24 @@ AnnounceResult announce(const torrent::TorrentFile& torrent_f) {
                       });
 
   if (res.status_code == 0 || res.status_code >= 400) {
-    throw std::runtime_error("could not announce to tracker");
+    auto logger = spdlog::get("custom");
+    logger->error("Could not announce to tracker");
+    return PeerResult::ERROR(PeerError::AnnounceError);
   }
-
   return parse_tracker_response(res.text);
 }
 
-AnnounceResult parse_tracker_response(const std::string& text) {
+PeerResult parse_tracker_response(const std::string& text) {
   AnnounceResult result;
 
   bencode::BencodeParser parser;
-  std::unique_ptr<bencode::BencodeValue> tree = parser.decode(text);
-  auto& dict = dynamic_cast<bencode::BencodeDict&>(*tree).value();
+  auto tree = parser.decode(text);
+  if(!tree.valid()){
+    auto logger = spdlog::get("custom");
+    logger->error("Parser error {}", static_cast<int>(tree.error()));
```

Comment:
Doesn't this print something like `Parser error 0`? I'd prefer to have the enum variant displayed as a string.